### PR TITLE
SDIT-2198 Correct sql for OffenderBookingReassigned and remove unused variables

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt [hunk 1]
```diff
@@ -471,66 +471,60 @@ class OracleToTopicIntTest : IntegrationTestBase() {
...
       private val newOffenderNo = "A5678BB"
-      private lateinit var newOffender: Offender
       private lateinit var bookingMoved: OffenderBooking
       private lateinit var movement: OffenderExternalMovement
 
       @BeforeEach
       fun setUp() {
         transaction {
           this.addLogger(StdOutSqlLogger)
-
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt:485]: just a tidy up in the test as not actually using the oldOfender and newOffender vars

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt [hunk 2]
```diff
@@ -561,6 +555,7 @@ class OracleToTopicIntTest : IntegrationTestBase() {
         assertThat(prisonerEvent.publishedAt).isCloseToUtcNow(within(10, ChronoUnit.SECONDS))
       }
     }
+    // Add more tests here and ensure different scenarios are covered
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt:558]: still work to do

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/repository/ExposeRepository.kt [hunk 3]
```diff
@@ -41,8 +41,8 @@ class ExposeRepository {
   fun getLastAdmissionDateForOffenderBooking(bookingId: Long): LocalDate? =
     OffenderBookings
       .join(OffenderExternalMovements, JoinType.INNER, additionalConstraint = {
-        (OffenderExternalMovements.offenderBookingId eq OffenderBookings.id)
-        (OffenderExternalMovements.type eq "ADM")
+        (OffenderExternalMovements.offenderBookingId eq OffenderBookings.id) and
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/repository/ExposeRepository.kt:44]: This is the actual change - sql wasn't quite correct

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**